extensions/emmet/src/abbreviationActions.ts

```diff
@@ -184,42 +184,70 @@ export function wrapIndividualLinesWithAbbreviation(args: any) {
 	}
 
 	const editor = vscode.window.activeTextEditor;
-	if (editor.selection.isEmpty) {
+	if (editor.selections.length === 1 && editor.selection.isEmpty) {
 		vscode.window.showInformationMessage('Select more than 1 line and try again.');
 		return;
 	}
+	if (editor.selections.find(x => x.isEmpty)) {
+		vscode.window.showInformationMessage('Select more than 1 line in each selection and try again.');
+		return;
+	}
+	let rangesToReplace: vscode.Range[] = [];
+	editor.selections.forEach(selection => {
+		let rangeToReplace: vscode.Range = selection.isReversed ? new vscode.Range(selection.active, selection.anchor) : selection;
+		if (!rangeToReplace.isSingleLine && rangeToReplace.end.character === 0) {
+			let previousLine = rangeToReplace.end.line - 1;
+			let lastChar = editor.document.lineAt(previousLine).text.length;
+			rangeToReplace = new vscode.Range(rangeToReplace.start, new vscode.Position(previousLine, lastChar));
+		}
+
+		rangeToReplace = ignoreExtraWhitespaceSelected(rangeToReplace, editor.document);
+		rangesToReplace.push(rangeToReplace);
+	});
 
 	const syntax = getSyntaxFromArgs({ language: editor.document.languageId });
 	if (!syntax) {
 		return;
 	}
 
 	const abbreviationPromise = (args && args['abbreviation']) ? Promise.resolve(args['abbreviation']) : vscode.window.showInputBox({ prompt: 'Enter Abbreviation' });
-	const lines = editor.document.getText(editor.selection).split('\n').map(x => x.trim());
 	const helper = getEmmetHelper();
 
 	return abbreviationPromise.then(inputAbbreviation => {
+		let expandAbbrInput: ExpandAbbreviationInput[] = [];
 		if (!inputAbbreviation || !inputAbbreviation.trim() || !helper.isAbbreviationValid(syntax, inputAbbreviation)) { return false; }
 
 		let extractedResults = helper.extractAbbreviationFromText(inputAbbreviation);
 		if (!extractedResults) {
 			return false;
 		}
+		rangesToReplace.forEach(rangeToReplace => {
```

Comment:
why not loop through `editor.selections` directly here, instead of doing so above and creating the temporary `rangesToReplace` array?